src/components/TopicElements/TopicElements.js

```diff
@@ -74,8 +74,12 @@ const getComponents = (defaultComponents, elementsToDisplay) =>
 function TopicElements({ history, appBaseUrl, loginUrl, staticFilesUrl }) {
   const ref = useRef(null);
   const dispatch = useDispatch();
-  const { activeTopic, layerService, map } = useSelector((state) => state.app);
+  const { t } = useTranslation();
+  const activeTopic = useSelector((state) => state.app.activeTopic);
+  const layerService = useSelector((state) => state.app.layerService);
+  const map = useSelector((state) => state.app.map);
```

Comment:
Is this for readability or performance? Just out of interest...

Reply:
Question of performance. It's bad practice to listen the all state.app object because it will rerender the component  every time the state change, not only when actievTopic, layerService amd map change.
 So listnening each property separetely will rerender the component only when the property change.